### ISSUE TITLE
Update to work with new outlets API

### ISSUE THREAD

**Issue by salzhrani** [ef4/liquid-fire#173]
anyone having trouble with the library and the latest ember beta, I get a a blank page when using liquid outlet.  not sure how to debug

**Comment by salzhrani** [comment]
With some debugging I found that execution get [here](https://github.com/ef4/liquid-fire/blob/master/app/views/liquid-outlet.js#L53) with `!oldView && !newView` evaluates to true and the view bails.

**Comment by ef4** [comment]
Thanks for reporting this and tracking it down. 

I will take a look too as soon as I can, unless you get it figured out first.

**Comment by salzhrani** [comment]
I could try, any pointers?

**Comment by ef4** [comment]
Your question made me sit and ponder for a minute, and then I realized that this probably broke due to my own change upstream in Ember (https://github.com/emberjs/ember.js/pull/10372). Outlets don't use `currentView` anymore.

I actually made that change because it's going to simplify liquid-fire and fix a bunch of our corner cases that used to be broken. But it does mean we need to update right away to take advantage of the new outlet implementation.

The TLDR is that outlets now get told what to render when their `setOutletState` function is called. That should replace our current observation of `currentView`. And liquid-outlet probably needs to extend OutletView (or CoreOutletView, although that may not be exposed to us at present) instead of just ContainerView.

It's a pretty significant change, and ideally we want to keep supporting both the old and new versions of Ember. If you want to go for it I'm happy to advise, otherwise hang in there and I will try to get to it soon.

**Comment by salzhrani** [comment]
I think it is better if you handled it since you are way more familiar with changes in both sides.

**Comment by brendanoh** [comment]
Is there a workaround for this?

**Comment by brendanoh** [comment]
I assume I am experiencing the same thing with the following output:
```html
<div id="ember440" class="ember-view liquid-container">
    <!---->
</div>
```

**Comment by ef4** [comment]
I just pushed a branch named beta-compat that works with beta and canary (but nothing older). Please give a whirl.

I may end up doing separate releases for Ember >= 1.11 and Ember < 1.11.

**Comment by brendanoh** [comment]
@ef4 works perfectly as usual. At least for all the transitions in my App.

Thanks again. Fantastic library.

**Comment by jgwhite** [comment]
@ef4 beta-compat appears to work perfectly on our app too. Thanks!

**Comment by miguelmota** [comment]
I spent hours trying to figure this out. `beta-compat` branch worked out!

**Comment by chrism** [comment]
@ef4 - works for me too, thanks for all your work :+1:

**Comment by miguelmota** [comment]
Should probably put this somewhere in the README.

**Comment by ef4** [comment]
Released 0.18.0 which is compatible with Ember beta and canary.

**Comment by zjrosen1** [comment]
@ef4 is this resolved in 0.19 

I'm getting the following error

`Uncaught TypeError: Cannot read property 'render' of undefined`

Which is coming from line 94 of ember-internals where the param state is undefined after a link transition

```javascript
setOutletState: function(state) {
    if (state.render && state.render.controller && !state._lf_model) {
      // This is a hack to compensate for Ember 1.0's remaining use of
      // mutability within the route state -- the controller is a
      // singleton whose model will keep changing on us. By locking it
      // down the first time we see the state, we can more closely
      // emulate ember 2.0 semantics.
      //
      // The Ember 2.0 component attributes shouldn't suffer this
      // problem and we can eventually drop the hack.
      state = Ember.copy(state);
      state._lf_model = get(state.render.controller, 'model');
    }
```

https://github.com/ef4/liquid-fire/blob/master/addon/ember-internals.js

**Comment by zjrosen1** [comment]
#239 Fixed above for me, thanks
